server/src/test/java/org/elasticsearch/index/query/InterceptedQueryBuilderWrapperTests.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class InterceptedQueryBuilderWrapperTests extends ESTestCase {
+
+    public void testInterceptedQueryBuildersAreUnwrapped() {
+        MatchAllQueryBuilder matchAllQueryBuilder = new MatchAllQueryBuilder();
```

Comment:
Shouldn't we test the actual rewriting mechanism? We're checking for equality here...

Reply:
Thanks for implementing it! 👍